experiments/config.toml

```diff
@@ -0,0 +1,74 @@
+title = "Mspush-Java's experiments."
+
+[owner]
+name = "David Ferreira Quaresma"
+email = "davidferreiraquaresma@gmail.com"
+
+[setup]
+num_vms = 4
+vcpus = 2
+ram_gb = 4
+java_version = "10.0.1"
+gcc="5.4.0" # compiled the libgc.so
+maven="3.3.9" # build all java project.
+go_version = "go1.10 linux/amd64"  # compiled the gci proxy binary.
+nginx_version = "nginx/1.14.0 (Ubuntu)"
+so_version = "Ubuntu 18.04 LTS"
+kernel_version = "4.15.0-29-generic"
+
+[execution]
+# exports: export commands which should be used to export common environment variables. Should use outdir=stateless for stateless experiments and outdir=stateful for stateful experiments.
+exports = 'export outdir="stateless"; export expdir=${PWD}'
+
+# killall: kill all processes started by the experiment.
+killall = 'for ip in $ip1 $ip2 $ip3 $ip4; do ssh -i ${sshkey} ubuntu@${ip} "killall java 2>/dev/null; rm gc.log shed_${expid}_${ip}.csv 2>/dev/null; killall mon.sh 2>/dev/null; killall gci-proxy 2>/dev/null; rm proxy.* 2>/dev/null;"; done'
+
+# setup: setup the environment (VMs/servers), guaranteeing that everything need to start the experiment is there. For example, copy files and sync repositories.
+setup = 'git clone http://github.com/gcinterceptor/gci-java && cd gci-java/ && ./build.sh; mv core/src/main/java/libgc.so ${expdir}; cd ${expdir}; git clone http://github.com/gcinterceptor/msgpush-java && cd msgpush-java && mvn clean package; cp experiments/mon.sh ${expdir}; cd target/ && mv msgpush-0.0.1-SNAPSHOT.jar msgpush-java.jar; cp msgpush-java.jar ${expdir}; cd ${expdir}; for ip in $ip1 $ip2 $ip3 $ip4; do scp -i ${sshkey} mon.sh ubuntu@${ip}:~/; scp -i ${sshkey} msgpush-java.jar ubuntu@${ip}:~/; scp -i ${sshkey} libgc.so ubuntu@${ip}:~/; done; go get -u github.com/gcinterceptor/gci-proxy; go get -u github.com/julienschmidt/httprouter; cd $GOPATH/src/github.com/gcinterceptor/gci-proxy && go build; for ip in $ip1 $ip2 $ip3 $ip4; do scp -i ${sshkey} gci-proxy ubuntu@${ip}:~/; done; cd ${expdir}'
+
+# load: trigger the load.
+load = 'ssh -i ${sshkey} ubuntu@${lb} "sudo rm /var/log/nginx/*.log; sudo service nginx restart; killall vegeta >/dev/null 2>&1; echo \"GET http://127.0.0.1/\" | ./vegeta attack -duration=120s -rate=30 | tee results.bin | ./vegeta report >client_${expid}.out 2>client_${expid}.err";'
+
+# fetch_results: fetches experiment results.
+fetch_results = 'scp -i ${sshkey} ubuntu@${lb}:~/*_${expid}* ${outdir}; for ip in $ip1 $ip2 $ip3 $ip4; do scp -i ${sshkey} ubuntu@${ip}:~/*_${expid}_${ip}* ${outdir}; done'
+
+# cleanup: killall processes, removes files and revert configuration changes done by the experiment run.
+cleanup = 'ssh -i ${sshkey} ubuntu@${lb} "rm *${expid}*"; for ip in $ip1 $ip2 $ip3 $ip4; do ssh -i ${sshkey} ubuntu@${ip} "killall java; killall mon.sh; killall gci-proxy 2>/dev/null; rm proxy.* 2>/dev/null; rm *${expid}_${ip}*"; done'
```

Comment:
cleanup should not kill processes (killall does)

> killall java; killall mon.sh; killall gci-proxy 2>/dev/null

Reply:
Done